src/Relude/Nub.hs

```diff
@@ -90,3 +97,27 @@ sortNub = Set.toList . Set.fromList
 -}
 unstableNub :: (Eq a, Hashable a) => [a] -> [a]
 unstableNub = HashSet.toList . HashSet.fromList
+
+{- | A generalized nub for any 'Foldable'. Not necessarily order preserving. -}
+foldNub :: (Foldable t, Ord a) => t a -> Set.Set a
+foldNub = foldr Set.insert Set.empty
```

Comment:
What do you think about alternative names like `foldToSet` and `foldToHashSet`?

Reply:
Yes, I like these names better.